src/main/java/io/strimzi/kafka/bridge/tracing/OpenTelemetryHandle.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.kafka.bridge.tracing;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.instrumentation.api.instrumenter.messaging.MessageOperation;
+import io.opentelemetry.instrumentation.kafkaclients.TracingProducerInterceptor;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import io.strimzi.kafka.bridge.config.BridgeConfig;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+import io.vertx.kafka.client.producer.KafkaProducerRecord;
+import org.apache.kafka.clients.producer.ProducerConfig;
+
+import java.util.Map;
+import java.util.Properties;
+
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.COMPONENT;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.KAFKA_SERVICE;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.OPENTELEMETRY_SERVICE_NAME_ENV_KEY;
+
+/**
+ * OpenTelemetry implementation of Tracing.
+ *
+ * Note: we use Vert.x OpenTelemetry extension to setup custom ContextStorageProvider:
```

Comment:
we know that it's happening because the Vert.x OpenTelemetry library is on the classpath and the OpenTelemetry framework is able to detect and configure it automatically. Would you mind to add a comment here about how it works please? I am sure we'll forget :-)